src/cogent3/core/profile.py

```diff
@@ -277,10 +277,19 @@ def __init__(self, data, motifs, row_indices=None):
         axis = 0 if self.array.ndim == 1 else 1
         validate_freqs_array(self.array, axis=axis)
 
-    def entropy(self):
-        """Shannon entropy per position using log2"""
+    def entropy_terms(self):
+        """Returns
+           -------
+           entropies : array
+                Has same dimension as self.array with
+                safe log operation applied.
+        """
         entropies = safe_p_log_p(self.array)
-        return entropies.sum(axis=1)
```

Comment:
because of how these will be used, you need to return a DictArray from the `entropy_terms` method, e.g.
```python
return self.template.wrap(entropies)
```